File: Medium/0075 - sortcolors/sortcolors.cpp
#include <vector>
using namespace std;

class Solution {
    public:
        void sortColors(vector<int>& nums)
        {
            int index0 = 0;
            int len = nums.size();
            int index2 = len - 1;
            int temp;

            int i = 0;
            while (i < len && i <= index2)
            {
                if (nums[i] == 0)
                {
                    if (index0 < i)
                    {
                        temp = nums[index0];
                        nums[index0] = 0;
                        nums[i] = temp;
                    } 
                    index0++;
                    i++;
                }
                else if (nums[i] == 2)
                {
                    temp = nums[index2];
                    nums[index2] = 2;
                    nums[i] = temp;
                    index2--;
                }
                else
                    i++;
            }
        }
};

File: Medium/0075 - sortcolors/sortcolors.py
class Solution(object):
    def sortColors(self, nums):
        index0 = 0
        lens = len(nums)
        index2 = lens - 1
        temp = 0

        i = 0
        while (i < lens and i <= index2) :
            if (nums[i] == 0) :
                if (index0 < i) :
                    temp = nums[index0]
                    nums[index0] = 0
                    nums[i] = temp
                index0 += 1
                i += 1
            elif (nums[i] == 2) :
                temp = nums[index2]
                nums[index2] = 2
                nums[i] = temp
                index2 -= 1
            else :
                i += 1